src/algebraic_geometry/projective_spectrum/scheme.lean

```diff
@@ -359,6 +380,217 @@ def to_Spec {f : A} (m : ℕ) (f_deg : f ∈ 𝒜 m) :
 
 end
 
+namespace from_Spec
+
+open graded_algebra set_like finset (hiding mk_zero)
+
+variables {𝒜} {f : A} {m : ℕ} {f_deg : f ∈ 𝒜 m}
+
+private meta def mem_tac : tactic unit :=
+let b : tactic unit :=
+  `[exact pow_mem_graded _ (submodule.coe_mem _) <|> exact nat_cast_mem_graded _ _] in
+b <|> `[by repeat { all_goals { apply graded_monoid.mul_mem } }; b]
+
+/--The function from `Spec A⁰_f` to `Proj|D(f)` is defined by `q ↦ {a | aᵢᵐ/fⁱ ∈ q}`, i.e. sending `q` a
```

Comment:
Whoa, didn't notice this line became overlong.

Reply:
reformatted